lms/djangoapps/grades/tasks.py

```diff
@@ -36,63 +56,59 @@ def recalculate_subsection_grade(
         only_if_higher (boolean): indicating whether grades should
             be updated only if the new raw_earned is higher than the
             previous value.
-        weighted_earned (float): the weighted points the learner earned on the
-            problem that triggered the update.
-        weighted_possible (float): the max weighted points the leaner could have
-            earned on the problem.
+        task_queued_time (serialized datetime): indicates when the task
+            was queued so that we can verify the underlying data update.
         score_deleted (boolean): indicating whether the grade change is
             a result of the problem's score being deleted.
     """
-    course_key = CourseLocator.from_string(course_id)
+    course_key = CourseLocator.from_string(kwargs['course_id'])
     if not PersistentGradesEnabledFlag.feature_enabled(course_key):
         return
 
     score_deleted = kwargs['score_deleted']
-    scored_block_usage_key = UsageKey.from_string(usage_id).replace(course_key=course_key)
+    scored_block_usage_key = UsageKey.from_string(kwargs['usage_id']).replace(course_key=course_key)
+    task_queued_time = datetime.strptime(kwargs['task_queued_time'], "%Y-%m-%d %H:%M:%S.%f").replace(tzinfo=pytz.UTC)
 
     # Verify the database has been updated with the scores when the task was
     # created. This race condition occurs if the transaction in the task
     # creator's process hasn't committed before the task initiates in the worker
     # process.
     if not _has_database_updated_with_new_score(
-            user_id, scored_block_usage_key, weighted_earned, weighted_possible, score_deleted,
+            kwargs['user_id'], scored_block_usage_key, task_queued_time, score_deleted,
     ):
-        raise _retry_recalculate_subsection_grade(
-            user_id, course_id, usage_id, only_if_higher, weighted_earned, weighted_possible, score_deleted,
-        )
+        raise _retry_recalculate_subsection_grade(**kwargs)
 
     _update_subsection_grades(
         course_key,
         scored_block_usage_key,
-        only_if_higher,
-        course_id,
-        user_id,
-        usage_id,
-        weighted_earned,
-        weighted_possible,
+        kwargs['only_if_higher'],
+        kwargs['course_id'],
+        kwargs['user_id'],
+        kwargs['usage_id'],
+        kwargs['task_queued_time'],
         score_deleted,
     )
 
 
 def _has_database_updated_with_new_score(
-        user_id, scored_block_usage_key, expected_raw_earned, expected_raw_possible, score_deleted,
+        user_id, scored_block_usage_key, task_queued_time, score_deleted,
 ):
     """
     Returns whether the database has been updated with the
     expected new score values for the given problem and user.
     """
-    score = get_score(user_id, scored_block_usage_key)
+    score = get_full_score(user_id, scored_block_usage_key)
```

Comment:
Damn, nvm. Turns out that `modified` value isn't always updated when a new submissions score is set. Back to the drawing board...